### PR TITLE
RUBY-2362 Add OCSP configurations to Evergreen


### PR DESCRIPTION
**Pull request by p-mongo**

### CODE REVIEW

**Comment by p-mongo** [comment]
https://jira.mongodb.org/browse/RUBY-2362

**Comment by p-mongo** [comment]
Ruby 2.3 and half of JRuby configurations are red, I'll fix them in follow-up PRs.

**Comment by p-mongo** [comment]
This PR also does not have the insecure/bypass verification configurations from https://github.com/mongodb/specifications/blob/master/source/ocsp-support/tests/README.rst#integration-tests-permutations-to-be-tested, I'll add them separately as well once 2.3/JRuby are fixed.

File: .evergreen/test-on-docker [hunk 1]
```diff
@@ -119,11 +119,17 @@ class Runner
...
       host_port = @options[:mongo_only] + i
       container_port = 27017 + i
       ['-p', "#{host_port}:#{container_port}"]
     end.flatten
+
+    if @env['OCSP_ALGORITHM'] && !@env['OCSP_VERIFIER']
+      args += %w(-p 8100:8100)
+    end
+
+    args
```
Comment (egiurleo) [inline .evergreen/test-on-docker:132]: Thanks for updating this as well 🙏

**Review by egiurleo — APPROVED** [review]
This is really thorough! Thank you!

**Pull request merged**